terraform/modules/kubernetes/monitoring.tf

```diff
@@ -115,3 +115,48 @@ resource "azurerm_monitor_metric_alert" "redis_memory" {
     ]
   }
 }
+data "azurerm_monitor_diagnostic_categories" "default" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+
+  resource_id = azurerm_postgresql_flexible_server.postgres-server[0].id
+}
+
+resource "azurerm_log_analytics_workspace" "pg_loganalytics" {
+  count = var.deploy_azure_backing_services ? 1 : 0
+  name                = "${azurerm_postgresql_flexible_server.postgres-server[0].name}-pgsql-logs"
+  location            = data.azurerm_resource_group.app-resource-group.location
+  resource_group_name = data.azurerm_resource_group.app-resource-group.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+resource "azurerm_monitor_diagnostic_setting" "pg_loganalytics_diagnostic" {
```

Comment:
"postgres" would be more explicit. There is no need to append "loganalytics_diagnostic"